thorlcr/thorutil/thmem.cpp

```diff
@@ -1235,31 +1235,41 @@ rowidx_t CThorSpillableRowArray::save(IFile &iFile, bool useCompression, const c
         nextCBI = nextCB->queryRecordNumber();
     }
     Owned<IExtRowWriter> writer = createRowWriter(&iFile, rowIf, rwFlags);
-    const void **rows = getBlock(n);
-    for (rowidx_t i=0; i < n; i++)
+    rowidx_t i=0;
+    try
     {
-        const void *row = rows[i];
-        assertex(row || allowNulls);
-        if (i == nextCBI)
+        const void **rows = getBlock(n);
+        while (i<n)
         {
-            writer->flush();
-            do
+            const void *row = rows[i];
+            assertex(row || allowNulls);
+            if (i == nextCBI)
             {
-                nextCB->filePosition(writer->getPosition());
-                if (cbCopy.ordinality())
+                writer->flush();
+                do
                 {
-                    nextCB = &cbCopy.pop();
-                    nextCBI = nextCB->queryRecordNumber();
+                    nextCB->filePosition(writer->getPosition());
+                    if (cbCopy.ordinality())
+                    {
+                        nextCB = &cbCopy.pop();
+                        nextCBI = nextCB->queryRecordNumber();
+                    }
+                    else
+                        nextCBI = RCIDXMAX; // indicating no more
                 }
-                else
-                    nextCBI = RCIDXMAX; // indicating no more
+                while (i == nextCBI); // loop as may be >1 IWritePosCallback at same pos
             }
-            while (i == nextCBI); // loop as may be >1 IWritePosCallback at same pos
+            rows[i++] = NULL;
```

Comment:
minor: this will leak row if there is an exception, but it will be recovered when the row manager is released.